zipline/finance/performance/period.py

```diff
@@ -242,6 +279,22 @@ def record_order(self, order):
     def handle_execution(self, txn):
         self.period_cash_flow += self._calculate_execution_cash_flow(txn)
 
+        asset = self.asset_finder.retrieve_asset(txn.sid)
+        if isinstance(asset, Future):
+            try:
+                old_price = self._payout_last_sale_prices[asset]
+                amount = self.position_tracker.positions[asset].amount
+                price = txn.price
+                cash_adj = (price - old_price) * asset.contract_multiplier * \
```

Comment:
This seems like code duplication with _get_payout_total()
Can the common ground be abstracted?